packages/catalog-model/src/validation/types.ts

```diff
@@ -24,4 +24,7 @@ export type Validators = {
   isValidAnnotationKey(value: unknown): boolean;
   isValidAnnotationValue(value: unknown): boolean;
   isValidTag(value: unknown): boolean;
+  isValidLinkUrl(value: unknown): boolean;
+  isValidLinkTitle(value: unknown): boolean;
+  isValidLinkIcon(value: unknown): boolean;
```

Comment:
This is an actual "breaking" change.

I'm not sure these are worth it. A valid url is a valid url - so having it in common is fine but maybe we can skip it here. And the title, the non-empty part of it is something we just typically slap in the schema because it makes absolutely no sense to allow the empty string. For the icons - well, not sure about that one, that's the only one where some marginal value could be had but, since it's just some nomenclature that people are free to choose as they see fit, and only within this very small scope of Backstage icons, I think the fixed rule of "same as metadata.name" is plenty forgiving.

Thoughts? I'd prefer just dropping these personally.

Reply:
Yep, that seems to make sense to me. This is my first time making modifications to the catalog model so wasn't entirely sure what should and shouldn't be included.

Reply:
Nah we don't. Could be improved wholesale across the project some time. We're switching to json-schema anyway so it'll be easy to slap checking regexes in there later 

https://github.com/backstage/backstage/pull/4277

(don't worry BTW about coordination, I'll let you know if we will collide, but I suspect you'll go in first anyway...)

Reply:
Or... Did I read that question right? :)

Reply:
Either way, #4277 doesn't seem to have many conflicts at all from my skim of the PR.